dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/interpretation/impl/DefaultInterpretationService.java

```diff
@@ -157,26 +213,151 @@ public void deleteInterpretation( Interpretation interpretation )
         return interpretationStore.getAllOrderedLastUpdated( first, max );
     }
 
+    @Override
+    public Set<User> getMentionedUsers( String text )
+    {
+        Set<User> users = new HashSet<>();
+        Matcher matcher = Pattern.compile( "(?:\\s|^)@([\\w+._-]+)" ).matcher( text );
```

Comment:
Can be made static and externalized, no need to compile every time

Reply:
Done here https://github.com/dhis2/dhis2-core/pull/1453/commits/8c29e1ac66cdf0f3abd2007e6f6c3c28a629c79f